HRCPP-421 Near cache eviction test and misc fixes

rigazilla opened the pull request:
This PR contains:

a test for the near cache evition
and some fixes on near cache tests

https://issues.jboss.org/browse/HRCPP-421

alanfx commented on xunit-test/NearCacheTest/CMakeLists.txt:
```diff
@@ -17,7 +17,7 @@ target_link_libraries(NearCacheMultiClientTest
+add_test(xunit_nearCacheTest xunit_nearCacheTest --gtest_output=xml:${PROJECT_BINARY_DIR}/test-output/)
 
 add_test (stop_server ${PYTHON_EXECUTABLE} ${CMAKE_ROOT_SOURCE_DIR}/test/bin/server_ctl.py stop)
 add_test (probe_port_stop ${PYTHON_EXECUTABLE} ${CMAKE_ROOT_SOURCE_DIR}/test/bin/probe_port.py localhost 11222 60 down)
```
`NearCacheEvictionTest` is not being executed

rigazilla replied:
@alanfx the tests fail randomically, I'm looking into.

rigazilla commented:
Changed to unsigned type the size of the near cache
fixed bug on near cache entry remove
added more robust checks in the test suite

@alanfx  be sure to rebuild the project from scratch, the new test needs new makefile from cmake

alanfx commented:
@rigazilla merging. More tests like failover can be added once the framework supports it.
